### PR TITLE
live hooks

### PR DESCRIPTION
**Pull request by eudaldgr**
First try to merge this with tinyramfs.
It only add support for overlayfs, but maybe works with aufs.

### CODE REVIEW

**Comment by illiliti** [comment]
Thanks for the PR. This change requires adding appropriate docs to man pages. Would you mind doing this ?

**Comment by eudaldgr** [comment]
I want to do, but before i would know your opinion or if i miss something or you don't want my hook. Now i will work with the man pages, if you agree.

**Comment by eudaldgr** [comment]
Well, man pages modified, what do you think about this PR?

**Comment by illiliti** [comment]
That's much better! There are some nitpicks that need to be fixed before i merge this

File: hooks/live/live [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (illiliti) [inline hooks/live/live:1]: ```# vim: set ft=sh:```

File: hooks/live/live.init [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# shellcheck disable=2154
+
+modprobe -a cdrom loop overlay squashfs 
+
+medium=/run/initramfs/medium
+system=/run/initramfs/system
+
+mkdir -p "$medium" "$system" /run/initramfs/overlayfs/work /run/initramfs/overlayfs/write
+
+[ -h "/dev/disk/by-$live_disk_type/$live_disk" ] || { sleep 5; }
```
Comment (illiliti) [inline hooks/live/live.init:13]: ```|| sleep 5```

File: hooks/live/live.init [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
```
Comment (illiliti) [inline hooks/live/live.init:1]: ```# vim: set ft=sh:```

**Comment by eudaldgr** [comment]
Well, in fact i don't use this hook anymore for my kiss livecd so i won't maintain this anymore, but i believe that there is a lot to do to improve it.

File: hooks/live/live (hunk 1 above)
Comment (eudaldgr) [reply hooks/live/live]: done

File: hooks/live/live.init (hunk 2 above)
Comment (eudaldgr) [reply hooks/live/live.init]: done

File: hooks/live/live.init (hunk 3 above)
Comment (eudaldgr) [reply hooks/live/live.init]: done

**Comment by wael444** [comment]
@eudaldgr , your kiss livecd is mounted from a cdrom.
tinyramfs does not include cdrom kernel drivers, how does it get mounted?

**Comment by eudaldgr** [comment]
> @eudaldgr , your kiss livecd is mounted from a cdrom. tinyramfs does not include cdrom kernel drivers, how does it get mounted?

The modules are copied from host kernel modules, you need to compile your kernel with `cdrom` enabled as a module.
This hook are unmantained, I made a live hook diskless for tinyramfs more simple, check it [here](https://github.com/eudaldgr/kiss-live/tree/master/hooks/tinyramfs)

**Comment by wael444** [comment]
my workaround was having a hook do `copy_kmod cdrom`.

**Comment by eudaldgr** [comment]
> my workaround was having a hook do `copy_kmod cdrom`.

When I opened this PR the implementation of `tinyramfs` was `copy_module $_mod` now it seems to be `copy_kmod $_mod`.

It needs improvement from conception.

**Comment by firasuke** [comment]
This is a very interesting PR for tinyramfs, and its approach for getting live boot to work is more distro-agnostic than what `kiss-live` is doing (which is having base tarballs and extracting them upon booting, which is slow...).

@eudaldgr I am wondering if you might consider opening it again, as I have written a similar proposal for [booster](https://github.com/anatol/booster/issues/232#issuecomment-1737913983), because apparently having an initramfs that works in live environments is hard, and I am interested in trying to get this working with your help.

**Comment by eudaldgr** [comment]
> This is a very interesting PR for tinyramfs, and its approach for getting live boot to work is more distro-agnostic than what `kiss-live` is doing (which is having base tarballs and extracting them upon booting, which is slow...).
> 
> @eudaldgr I am wondering if you might consider opening it again, as I have written a similar proposal for [booster](https://github.com/anatol/booster/issues/232#issuecomment-1737913983), because apparently having an initramfs that works in live environments is hard, and I am interested in trying to get this working with your help.

I am no longer working on this, feel free to (re-)open a similar pull request yourself if you are so interested.

**Pull request closed without merging**